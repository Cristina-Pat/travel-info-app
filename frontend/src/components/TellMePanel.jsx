import { useState} from "react";
import { useNavigate } from "react-router-dom";

const TellMePanel = () => {

  let [inputValue, setInputValue] = useState("");
  let navigate = useNavigate();
  
  function handleOnClick() {
    navigate("/weather?location=" + inputValue);
  }

  let handleChange = function (e) {
    setInputValue(e.target.value);
  };

    return (
        <div className="tell-me d-flex justify-content-center align-items-center">
          <form className="location-input d-block" role="search" >
            <h2 className="mb-3">Tell me about...</h2>
            <input
              className="form-control me-2 "
              onChange={handleChange}
              type="search"
              placeholder="Location name..."
              aria-label="Search"
              name="location"
            />
            <div className="text-center">
              <button className="location-search-btn btn btn-outline-primary mt-3" onClick={handleOnClick} type="submit">
                Search
              </button>
            </div>
          </form>
        </div>
    );
};

export default TellMePanel;