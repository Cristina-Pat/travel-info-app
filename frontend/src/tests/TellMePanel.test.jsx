import { render } from '@testing-library/react';
import TellMePanel from '../components/TellMePanel';

test(`TellMePanel matches snapshot`, () => {
  const component = render(<TellMePanel />);

  expect(component).toMatchSnapshot();
});