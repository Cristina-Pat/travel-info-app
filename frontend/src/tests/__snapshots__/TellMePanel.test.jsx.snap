// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`TellMePanel matches snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="tell-me d-flex justify-content-center align-items-center"
      >
        <form
          action="/weather"
          class="location-input d-block"
          role="search"
        >
          <h2
            class="mb-3"
          >
            Tell me about...
          </h2>
          <input
            aria-label="Search"
            class="form-control me-2 "
            name="location"
            placeholder="Location name..."
            type="search"
          />
          <div
            class="text-center"
          >
            <button
              class="location-search-btn btn btn-outline-primary mt-3"
              type="submit"
            >
              Search
            </button>
          </div>
        </form>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="tell-me d-flex justify-content-center align-items-center"
    >
      <form
        action="/weather"
        class="location-input d-block"
        role="search"
      >
        <h2
          class="mb-3"
        >
          Tell me about...
        </h2>
        <input
          aria-label="Search"
          class="form-control me-2 "
          name="location"
          placeholder="Location name..."
          type="search"
        />
        <div
          class="text-center"
        >
          <button
            class="location-search-btn btn btn-outline-primary mt-3"
            type="submit"
          >
            Search
          </button>
        </div>
      </form>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
